app/models/comment_entry.rb

```diff
@@ -1,4 +1,10 @@
 class CommentEntry < ApplicationRecord
-  belongs_to :author
   belongs_to :post
+  belongs_to :author, class_name: 'User'
+
+  after_create :update_post_comments_counter
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comment_entries.count)
+  end
```

Comment:
:red_circle: [REQUIRED] The method you provided, update_post_comments_counter, does not increment the comments_counter attribute. The update method takes a hash as its argument, and the key-value pair comments_counter: post.comment_entries.count tells the update method to update the comments_counter attribute to the current number of comments on the post.

However, this method does not actually increment the comments_counter attribute. It simply sets the value of the attribute to the current number of comments. If the post has no comments, then the value of the comments_counter attribute will be 0.

The better approach to increment the comments_counter attribute by 1 every time a new comment is added is to use the increment! method. The increment! method will increment the value of the attribute by 1, and it will also save the changes to the database.

Here is the corrected code:

```
def update_post_comments_counter
  # Increment the comments_counter attribute by 1.
  post.increment!(:comments_counter)
end
```

This code will first find the Post object. Then, it will increment the comments_counter attribute by 1. Finally, it will save the changes to the database

Reply:
@bienvenuushindi done